module_utils/network_lsr/argument_validator.py

```diff
@@ -413,6 +414,73 @@ def _validate_post(self, value, name, result):
 
         return result
 
+class ArgValidatorIPRule(ArgValidatorDict):
+    def __init__(self, name, family=None, required=False, default_value=None):
+        ArgValidatorDict.__init__(
+            self,
+            name,
+            required,
+            nested=[
+                ArgValidatorIP(
+                    "from", family=family, required=False, plain_address=False
+                ),
+                ArgValidatorNum("from_prefix", default_value=32, val_min=0, val_max=32),
+                ArgValidatorIP(
+                    "to", family=family, required=False, plain_address=False
+                ),
+                ArgValidatorNum("to_prefix", default_value=32, val_min=0, val_max=32),
+                ArgValidatorStr(
+                    "iif", default_value=None, enum_values=listdir('/sys/class/net/')
+                ),
+                ArgValidatorStr(
+                    "oif", default_value=None, enum_values=listdir('/sys/class/net/')
+                ),
+                ArgValidatorNum(
+                    "tos", default_value=None, val_min=0, val_max=255
+                ),
+                ArgValidatorNum(
+                    "dsfield", default_value=None, val_min=0, val_max=255
+                ),
+                ArgValidatorNum(
+                    "fwmark", default_value=None, val_min=0, val_max=65535
+                ),
+                # Not implemented
+                # ArgValidatorRange(
+                #     "uidrange", default_value=None, val_min=0, val_max=65535
+                # ),
+                ArgValidatorNum(
+                    "ipproto", default_value=None, val_min=0, val_max=255
+                ),
+                # ArgValidatorRange(
+                #     "sport", default_value=None, val_min=0, val_max=65535
+                # ),
+                # ArgValidatorDport(
+                #     "dport", default_value=None, val_min=0, val_max=65535
+                # ),
+                ArgValidatorNum(
+                    "priority", default_value=None, val_min=0, val_max=32767
+                ),
+                ArgValidatorStr(
+                    "table", default_value="main"
+                ),
+                ArgValidatorNum(
+                    "protocol", default_value=None, val_min=1, val_max=255
+                ),
+                ArgValidatorNum(
+                    "suppress_prefixlength", default_value=None, val_min=0, val_max=32
```

Comment:
the maximum depends on the address family. This has to be checked in `_validate_post()`.